xray/test/test_plot.py

```diff
@@ -524,3 +597,185 @@ def test_seaborn_palette_needs_levels(self):
                 self.plotmethod(cmap='husl')
         except ImportError:
             pass
+
+
+class TestFacetGrid(PlotTestCase):
+
+    def setUp(self):
+        d = easy_array((10, 15, 3))
+        self.darray = DataArray(d, dims=['y', 'x', 'z'])
+        self.g = xplt.FacetGrid(self.darray, col='z')
+
+    def test_no_args(self):
+        self.g.map_dataarray(xplt.contourf, 'x', 'y')
+
+        # Don't want colorbar labeled with 'None'
+        alltxt = text_in_fig()
+        self.assertNotIn('None', alltxt)
+
+        for ax in self.g:
+            self.assertTrue(ax.has_data())
+
+            # default font size should be small
+            fontsize = ax.title.get_size()
+            self.assertLessEqual(fontsize, 12)
+
+    def test_names_appear_somewhere(self):
+        self.darray.name = 'testvar'
+        self.g.map_dataarray(xplt.contourf, 'x', 'y')
+        for i, ax in enumerate(self.g):
+            self.assertEqual('z = {0}'.format(i), ax.get_title())
+
+        alltxt = text_in_fig()
+        self.assertIn(self.darray.name, alltxt)
+        for label in ['x', 'y']:
+            self.assertIn(label, alltxt)
+
+    def test_text_not_super_long(self):
+        self.darray.coords['z'] = [100 * letter for letter in 'abc']
+        g = xplt.FacetGrid(self.darray, col='z')
+        g.map_dataarray(xplt.contour, 'x', 'y')
+        alltxt = text_in_fig()
+        maxlen = max(len(txt) for txt in alltxt)
+        self.assertLess(maxlen, 50)
+
+        t0 = g.axes[0, 0].get_title()
+        self.assertTrue(t0.endswith('...'))
+
+    def test_colorbar(self):
+        vmin = self.darray.values.min()
+        vmax = self.darray.values.max()
+        expected = np.array((vmin, vmax))
+
+        self.g.map_dataarray(xplt.imshow, 'x', 'y')
+
+        for image in plt.gcf().findobj(mpl.image.AxesImage):
+            clim = np.array(image.get_clim())
+            self.assertTrue(np.allclose(expected, clim))
+
+        # There's only one colorbar
+        cbar = plt.gcf().findobj(mpl.collections.QuadMesh)
+        self.assertEqual(1, len(cbar))
+        
+    def test_empty_cell(self):
+        g = xplt.FacetGrid(self.darray, col='z', col_wrap=2)
+        g.map_dataarray(xplt.imshow, 'x', 'y')
+        
+        bottomright = g.axes[-1, -1]
+        self.assertFalse(bottomright.has_data())
+        self.assertFalse(bottomright.get_visible())
+
+    def test_norow_nocol_error(self):
+        with self.assertRaisesRegexp(ValueError, r'[Rr]ow'):
+            xplt.FacetGrid(self.darray)
+
+    def test_groups(self):
+        self.g.map_dataarray(xplt.imshow, 'x', 'y')
+        upperleft_dict = self.g.name_dicts[0, 0]
+        upperleft_array = self.darray[upperleft_dict]
+        z0 = self.darray.isel(z=0)
+
+        self.assertDataArrayEqual(upperleft_array, z0)
+        # Not sure if we need to expose this in this way
```

Comment:
Hmm. It does seem like it could be useful to save the artists generated by the mapping function -- these are hard to recover afterwards. Something like `self.artists`? On the other hand, that probably doesn't make much sense if you call map/map_dataarray multiple times, so maybe not.